syntax = "proto3";

package cm.proto;

// Emulates the ProcessingMetaData struct
message ProcessingMetaData {
  bool captured_from_real_device = 1;
  int64 capture_timestamp_ms = 2;
  int64 capture_from_this_device_since_ms = 3;
  uint32 frame_seq_num = 4;
  int32 processing_unit_idx = 5;
  float change_rate = 6;
  float fps = 7;
}


message FrameMsg {
  // If true, 'frame' contains raw Mat data (rows/cols/type header + pixels).
  // If false, 'frame' likely contains a compressed JPEG/PNG buffer.
  bool is_cv_mat = 1;
  // Binary data for the image.
  bytes frame = 2;
  ProcessingMetaData meta_data = 3;
}